python-client/giskard/scanner/llm/business_detector.py

```diff
@@ -0,0 +1,136 @@
+        results = pd.DataFrame.from_records(
+            [result.__dict__ for result in validate_test_case_with_reason(model, test_case, dataset.df, predictions)]
+        )
+        failed = results["score"] < 3
+        failed_count = len([result for result in failed if result])
+        metric = failed_count / len(predictions)
+        maybe_print(f"Results: {metric} ({failed_count})", verbose=verbose)
+
+        df_with_pred_and_test_results = pd.concat([df_with_pred, results[["reason", "tip"]]], axis=1)
+
+        if failed_count > 0:
+            maybe_print("Test failed", verbose)
+
+            issues.append(
+                Issue(
+                    model,
+                    dataset,
+                    level=IssueLevel.MAJOR if metric >= threshold else IssueLevel.MINOR,
+                    group=IssueGroup(name=issue.name, description=issue.description),
+                    description=f"For the test '{test_case}', we found that {metric * 100:.2f} of the generated answers does not respect it.",
+                    meta={
+                        "domain": test_case,
+                        "metric_value": metric,
+                        "test_case": test_case,
+                        "deviation": f"{round(metric * 100, 2)}% of generated inputs does not respect the test",
+                        "hide_index": True,
+                    },
+                    examples=df_with_pred_and_test_results[failed],
+                    tests=_generate_business_test,
+                )
+            )
+
+    return issues
+
+
+@detector("llm_business", tags=["business", "llm", "generative", "text_generation"])
+class LLMBusinessDetector:
+    def __init__(self, threshold: float = 0.6, num_samples=10, num_tests=3):
+        self.threshold = threshold
+        self.num_samples = num_samples
+        self.num_tests = num_tests
+
+    def run(self, model: LangchainModel, _: Dataset, verbose=True, **kwargs) -> Sequence[Issue]:
+
+        issues = []
+
+        for issue in LLM_ISSUE_CATEGORIES:
+            try:
+                test_cases = (
+                    issue.issue_generator(self.num_tests)
+                    .run_and_parse(model_name=model.meta.name, model_description=model.meta.description)
+                    .assertions[: self.num_tests]
+                )
+                maybe_print(f"Generated tests: {test_cases}", verbose=verbose)
+
+                potentially_failing_inputs = (
+                    issue.input_generator(self.num_samples)
+                    .run_and_parse_with_prompt(
+                        model_name=model.meta.name,
+                        model_description=model.meta.description,
+                        variables=model.meta.feature_names,
+                        generated_tests=test_cases,
+                    )
+                    .input[: self.num_samples]
+                )
+
+                potentially_failing_dataset = Dataset(
```

Comment:
Can we say that "potentially_failing_dataset" is just the "evaluation dataset"?

Reply:
Yes its the evaluation dataset your correct